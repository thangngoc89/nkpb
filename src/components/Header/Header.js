import React, { PropTypes } from 'react'
import { TitleBar } from 'react-desktop/windows'
import styles from './index.css'

const Header = (props) => (
  <div className={styles.header}>
    <span className={styles.title}>Dental Management</span>

    <TitleBar
      title=" "
      controls
      isMaximized={props.isMaximized}
      theme={props.theme}
      background={props.background}
      onCloseClick={props.close}
      onMinimizeClick={() => props.minimize(true)}
      onMaximizeClick={() => props.maximize(true)}
      onRestoreDownClick={() => props.unmaximize(true)}
    />
  </div>
)

Header.propTypes = {
  isMinimized: PropTypes.bool.isRequired,
  isMaximized: PropTypes.bool.isRequired,
  close: PropTypes.func.isRequired,
  maximize: PropTypes.func.isRequired,
  unmaximize: PropTypes.func.isRequired,
  minimize: PropTypes.func.isRequired,
  restore: PropTypes.func.isRequired,
  theme: PropTypes.string,
  background: PropTypes.string,
}

Header.defaultProps = {
  background: 'transparent',
  theme: 'light',
}

export default Header
