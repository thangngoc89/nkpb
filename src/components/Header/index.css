.header {
  position: fixed;
  top: 0;
  z-index: 100;
  width: 100vw;
  height: 32px;
  -webkit-user-select: none;
  -webkit-app-region: drag;
  background: rgba(0, 0, 0, 0.2);
}

.header a {
  z-index: 1000;
}

.title {
  width: 100vw;
  display: block;
  top: 10px;
  position: absolute;
  text-align: center;
}
