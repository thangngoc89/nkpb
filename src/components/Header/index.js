import React, { Component, PropTypes } from 'react'
import { TitleBar } from 'react-desktop/windows'
import { remote } from 'electron'
import styles from './index.css'

export default class Header extends Component {
  static propTypes = {
    isMinimized: PropTypes.bool.isRequired,
    isMaximized: PropTypes.bool.isRequired,
    isClosed: PropTypes.bool.isRequired,
    close: PropTypes.func.isRequired,
    maximize: PropTypes.func.isRequired,
    unmaximize: PropTypes.func.isRequired,
    minimize: PropTypes.func.isRequired,
    restore: PropTypes.func.isRequired,
    theme: PropTypes.string,
    background: PropTypes.string,
  }

  static defaultProps = {
    background: 'transparent',
    theme: 'light',
  }

  componentDidMount() {
    const browserWindow = remote.getCurrentWindow()
    browserWindow.on('minimize', () => this.props.minimize())
    browserWindow.on('maximize', () => this.props.maximize())
    browserWindow.on('restore', () => this.props.restore())
    browserWindow.on('unmaximize', () => this.props.unmaximize())
  }

  render() {
    return (
      <div className={styles.header}>
        <span className={styles.title}>Dental Management</span>

        <TitleBar
          title=" "
          controls
          isMaximized={this.props.isMaximized}
          theme={this.props.theme}
          background={this.props.background}
          onCloseClick={this.props.close}
          onMinimizeClick={() => this.props.minimize(true)}
          onMaximizeClick={() => this.props.maximize(true)}
          onRestoreDownClick={() => this.props.unmaximize(true)}
        />
      </div>
    )
  }
}
