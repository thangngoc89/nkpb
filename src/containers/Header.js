import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import Header from '../components/Header/'
import { actions as HeaderActions } from '../redux/modules/header'

function mapStateToProps({ header }) {
  return { ...header }
}

export default connect(
  mapStateToProps,
  (dispatch) => bindActionCreators(HeaderActions, dispatch),
)(Header)
