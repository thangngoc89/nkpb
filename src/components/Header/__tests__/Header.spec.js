import React from 'react'
import renderer from 'react-test-renderer'
import Header from '../Header'

const noop = () => {}
const props = {
  isMaximized: false,
  isMinimized: false,
  close: noop,
  maximize: noop,
  unmaximize: noop,
  minimize: noop,
  restore: noop,
}

describe('Header', () => {
  it('render with default props', () => {
    const component = renderer.create(
      <Header {...props} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('render with custom backgrund and theme', () => {
    const component = renderer.create(
      <Header background="black" theme="dark" {...props} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
