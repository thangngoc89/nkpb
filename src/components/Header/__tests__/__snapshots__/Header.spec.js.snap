exports[`Header render with custom backgrund and theme 1`] = `
<div
  className={undefined}>
  <span
    className={undefined}>
    Dental Management
  </span>
  <div
    style={
      Object {
        "WebkitAppRegion": "drag",
        "WebkitUserSelect": "none",
        "alignItems": "center",
        "appRegion": "drag",
        "background": "black",
        "cursor": "default",
        "display": "flex",
        "height": "31px",
        "userSelect": "none",
        "visibility": "visible",
        "width": "100%"
      }
    }>
    <div
      style={
        Object {
          "WebkitUserSelect": "none",
          "color": "#ffffff",
          "cursor": "default",
          "flex": 1,
          "fontFamily": "\"Segoe UI\", \"Arial\"",
          "fontSize": "12px",
          "paddingLeft": "12px",
          "userSelect": "none"
        }
      }>
       
    </div>
    <div
      style={
        Object {
          "WebkitUserSelect": "none",
          "cursor": "default",
          "display": "flex",
          "height": "32px",
          "userSelect": "none"
        }
      }>
      <a
        data-radium={true}
        onClick={[Function onMinimizeClick]}
        onKeyDown={[Function anonymous]}
        onKeyUp={[Function anonymous]}
        onMouseDown={[Function anonymous]}
        onMouseEnter={[Function anonymous]}
        onMouseLeave={[Function anonymous]}
        style={
          Object {
            "WebkitAppRegion": "no-drag",
            "WebkitUserSelect": "none",
            "alignItems": "center",
            "appRegion": "no-drag",
            "cursor": "default",
            "display": "flex",
            "height": "100%",
            "justifyContent": "center",
            "lineHeight": 0,
            "userSelect": "none",
            "width": "46px"
          }
        }
        title="Minimize">
        <svg
          data-radium={true}
          style={
            Object {
              "height": "10px",
              "width": "10px"
            }
          }
          viewBox="0 0 10.2 1"
          x="0px"
          y="0px">
          <rect
            fill="#ffffff"
            height="1"
            width="10.2" />
        </svg>
      </a>
      <a
        data-radium={true}
        onClick={[Function onMaximizeClick]}
        onKeyDown={[Function anonymous]}
        onKeyUp={[Function anonymous]}
        onMouseDown={[Function anonymous]}
        onMouseEnter={[Function anonymous]}
        onMouseLeave={[Function anonymous]}
        style={
          Object {
            "WebkitAppRegion": "no-drag",
            "WebkitUserSelect": "none",
            "alignItems": "center",
            "cursor": "default",
            "display": "flex",
            "height": "100%",
            "justifyContent": "center",
            "lineHeight": 0,
            "userSelect": "none",
            "width": "46px"
          }
        }
        title="Maximize">
        <svg
          data-radium={true}
          style={
            Object {
              "height": "10px",
              "width": "10px"
            }
          }
          viewBox="0 0 10.2 10.1"
          x="0px"
          y="0px">
          <path
            d="M0,0v10.1h10.2V0H0z M9.2,9.2H1.1V1h8.1V9.2z"
            fill="#ffffff" />
        </svg>
      </a>
      <a
        data-radium={true}
        onClick={[Function noop]}
        onKeyDown={[Function anonymous]}
        onKeyUp={[Function anonymous]}
        onMouseDown={[Function anonymous]}
        onMouseEnter={[Function anonymous]}
        onMouseLeave={[Function anonymous]}
        style={
          Object {
            "WebkitAppRegion": "no-drag",
            "WebkitUserSelect": "none",
            "alignItems": "center",
            "cursor": "default",
            "display": "flex",
            "height": "100%",
            "justifyContent": "center",
            "lineHeight": 0,
            "userSelect": "none",
            "width": "46px"
          }
        }
        title="Close">
        <svg
          data-radium={true}
          style={
            Object {
              "height": "10px",
              "width": "10px"
            }
          }
          viewBox="0 0 10.2 10.2"
          x="0px"
          y="0px">
          <polygon
            fill="#ffffff"
            points="10.2,0.7 9.5,0 5.1,4.4 0.7,0 0,0.7 4.4,5.1 0,9.5 0.7,10.2 5.1,5.8 9.5,10.2 10.2,9.5 5.8,5.1 " />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`Header render with default props 1`] = `
<div
  className={undefined}>
  <span
    className={undefined}>
    Dental Management
  </span>
  <div
    style={
      Object {
        "WebkitAppRegion": "drag",
        "WebkitUserSelect": "none",
        "alignItems": "center",
        "appRegion": "drag",
        "background": "transparent",
        "cursor": "default",
        "display": "flex",
        "height": "31px",
        "userSelect": "none",
        "visibility": "visible",
        "width": "100%"
      }
    }>
    <div
      style={
        Object {
          "WebkitUserSelect": "none",
          "color": "#000000",
          "cursor": "default",
          "flex": 1,
          "fontFamily": "\"Segoe UI\", \"Arial\"",
          "fontSize": "12px",
          "paddingLeft": "12px",
          "userSelect": "none"
        }
      }>
       
    </div>
    <div
      style={
        Object {
          "WebkitUserSelect": "none",
          "cursor": "default",
          "display": "flex",
          "height": "32px",
          "userSelect": "none"
        }
      }>
      <a
        data-radium={true}
        onClick={[Function onMinimizeClick]}
        onKeyDown={[Function anonymous]}
        onKeyUp={[Function anonymous]}
        onMouseDown={[Function anonymous]}
        onMouseEnter={[Function anonymous]}
        onMouseLeave={[Function anonymous]}
        style={
          Object {
            "WebkitAppRegion": "no-drag",
            "WebkitUserSelect": "none",
            "alignItems": "center",
            "appRegion": "no-drag",
            "cursor": "default",
            "display": "flex",
            "height": "100%",
            "justifyContent": "center",
            "lineHeight": 0,
            "userSelect": "none",
            "width": "46px"
          }
        }
        title="Minimize">
        <svg
          data-radium={true}
          style={
            Object {
              "height": "10px",
              "width": "10px"
            }
          }
          viewBox="0 0 10.2 1"
          x="0px"
          y="0px">
          <rect
            fill="#000000"
            height="1"
            width="10.2" />
        </svg>
      </a>
      <a
        data-radium={true}
        onClick={[Function onMaximizeClick]}
        onKeyDown={[Function anonymous]}
        onKeyUp={[Function anonymous]}
        onMouseDown={[Function anonymous]}
        onMouseEnter={[Function anonymous]}
        onMouseLeave={[Function anonymous]}
        style={
          Object {
            "WebkitAppRegion": "no-drag",
            "WebkitUserSelect": "none",
            "alignItems": "center",
            "cursor": "default",
            "display": "flex",
            "height": "100%",
            "justifyContent": "center",
            "lineHeight": 0,
            "userSelect": "none",
            "width": "46px"
          }
        }
        title="Maximize">
        <svg
          data-radium={true}
          style={
            Object {
              "height": "10px",
              "width": "10px"
            }
          }
          viewBox="0 0 10.2 10.1"
          x="0px"
          y="0px">
          <path
            d="M0,0v10.1h10.2V0H0z M9.2,9.2H1.1V1h8.1V9.2z"
            fill="#000000" />
        </svg>
      </a>
      <a
        data-radium={true}
        onClick={[Function noop]}
        onKeyDown={[Function anonymous]}
        onKeyUp={[Function anonymous]}
        onMouseDown={[Function anonymous]}
        onMouseEnter={[Function anonymous]}
        onMouseLeave={[Function anonymous]}
        style={
          Object {
            "WebkitAppRegion": "no-drag",
            "WebkitUserSelect": "none",
            "alignItems": "center",
            "cursor": "default",
            "display": "flex",
            "height": "100%",
            "justifyContent": "center",
            "lineHeight": 0,
            "userSelect": "none",
            "width": "46px"
          }
        }
        title="Close">
        <svg
          data-radium={true}
          style={
            Object {
              "height": "10px",
              "width": "10px"
            }
          }
          viewBox="0 0 10.2 10.2"
          x="0px"
          y="0px">
          <polygon
            fill="#000000"
            points="10.2,0.7 9.5,0 5.1,4.4 0.7,0 0,0.7 4.4,5.1 0,9.5 0.7,10.2 5.1,5.8 9.5,10.2 10.2,9.5 5.8,5.1 " />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
